fl4health/strategies/feddg_ga_strategy.py

```diff
@@ -0,0 +1,382 @@
+                will still be sampled. Defaults to 1.0.
+            min_fit_clients : int, optional
+                Minimum number of clients used during training. Defaults to 2.
+            min_evaluate_clients : int, optional
+                Minimum number of clients used during validation. Defaults to 2.
+            min_available_clients : int, optional
+                Minimum number of total clients in the system. Defaults to 2.
+            evaluate_fn :
+                Optional[
+                    Callable[[int, NDArrays, Dict[str, Scalar]],
+                    Optional[Tuple[float, Dict[str, Scalar]]]]
+                ]
+                Optional function used for validation. Defaults to None.
+            on_fit_config_fn : Callable[[int], Dict[str, Scalar]], optional
+                Function used to configure training. Defaults to None.
+            on_evaluate_config_fn : Callable[[int], Dict[str, Scalar]], optional
+                Function used to configure validation. Defaults to None.
+            accept_failures : bool, optional
+                Whether or not accept rounds containing failures. Defaults to True.
+            initial_parameters : Parameters, optional
+                Initial global model parameters.
+            fit_metrics_aggregation_fn : Optional[MetricsAggregationFn]
+                Metrics aggregation function, optional.
+            evaluate_metrics_aggregation_fn : Optional[MetricsAggregationFn]
+                Metrics aggregation function, optional.
+            fairness_metric : FairnessMetric, optional.
+                The metric to evaluate the local model of each client against the global model in order to
+                determine their adjustment weight for aggregation. Can be set to any default metric in
+                FairnessMetricType or set to use a custom metric. Optional, default is
+                FairnessMetric(FairnessMetricType.LOSS).
+            weight_step_size : float
+                The step size to determine the magnitude of change for the adjustment weight.
+                Optional, default is 0.2.
+        """
+        super().__init__(
+            fraction_fit=fraction_fit,
+            fraction_evaluate=fraction_evaluate,
+            min_fit_clients=min_fit_clients,
+            min_evaluate_clients=min_evaluate_clients,
+            min_available_clients=min_available_clients,
+            evaluate_fn=evaluate_fn,
+            on_fit_config_fn=on_fit_config_fn,
+            on_evaluate_config_fn=on_evaluate_config_fn,
+            accept_failures=accept_failures,
+            initial_parameters=initial_parameters,
+            fit_metrics_aggregation_fn=fit_metrics_aggregation_fn,
+            evaluate_metrics_aggregation_fn=evaluate_metrics_aggregation_fn,
+        )
+
+        if fairness_metric is None:
+            self.fairness_metric = FairnessMetric(FairnessMetricType.LOSS)
```

Comment:
We can make the `fairness_metric` input default to this fairness metric by default in the arguments above instead of None and then just do `self.fairness_metric = fairness_metric` here I think?

Reply:
hmmm...actually, because its a class that's a bad idea 😂

Reply:
Yep! The most annoying python bug/gotcha they insist on not fixing!!!